src/DistributedMetadata/MetadataService.cpp

```diff
@@ -266,6 +267,8 @@ void MetadataService::startup()
     /// Producer settings
     kctx.request_required_acks = conf.request_required_acks;
     kctx.request_timeout_ms = conf.request_timeout_ms;
+    /// Compression settings
+    kctx.client_side_compression = config.getBool(conf.key_prefix + COMPRESSION_KEY , false);
```

Comment:
In `config.xml`, we shall update it with `<client_compression> true </client_compression>` for catalogs topic ?

Reply:
If client_compression is enabled for a topic, during topic creation `KafkaWAL::create(...)`, we need disable topic level compression